### PR TITLE
Add support for ignoring languages

### PR DESCRIPTION
**Pull request by icholy**
Relates to #82

### CODE REVIEW

**Comment by icholy** [comment]
After using this for about 5 minutes I realized that what I actually want is to blacklist languages.

**Comment by icholy** [comment]
I need this so I can use [json-formatter](https://chrome.google.com/webstore/detail/json-formatter/bcjindcccaagfpapjjmafapmmgkkhgoa?hl=en) for json files.

File: js/background.js [hunk 1]
```diff
@@ -77,7 +77,8 @@
     theme: 'sunburst',
     font: 'Inconsolata',
     fontSize: 'medium',
-    lineNumbers: true
+    lineNumbers: true,
+    languageBlacklist: ''
```
Comment (tsenart) [inline js/background.js]: I'd generalise this by making the value a regular expression that's matched against the URL. This way we can not only skip specific languages but whole domains. What do you think? I'd call it, `ignore`.

Comment (icholy) [inline js/background.js]: The problem is that the language is often determined by the content-type. In those cases, url filtering isn't enough.

Comment (tsenart) [inline js/background.js]: Yes, you're right. Without wanting to overly complicate things, for this PR in particular, but allowing for domain blacklisting to gracefully extend this work, what would you think of:

```js
blacklists: {
    languages: ['javascript', 'ruby'],
    domains: ['github.com']
}
```

Comment (icholy) [inline js/background.js]: I have no problems with that. Did you want to rename it to `ignore`?

``` js
ignore: {
    languages: ['javascript', 'ruby'],
    domains: ['github.com']
}
```

Comment (icholy) [inline js/background.js]: With the current way options are handled, I'm not sure how to create that nested structure.

**Comment by icholy** [comment]
I've renamed it from "blacklist" to "ignore".

**Comment by tsenart** [comment]
OK, I was testing this locally and I'm not super happy with the textarea interface to configure this. The user wouldn't know what to type: Javascript, JavaScript, JS, Js, js. I think we should either create checkboxes for each language (easy) or enable auto-completion of the language names (harder).

**Comment by icholy** [comment]
Something like this? http://www.jqueryscript.net/demo/jQuery-jQuery-UI-Plugin-For-Simle-Tokenized-Autocomplete-Autocomplete-Multiselect/

**Comment by tsenart** [comment]
Yeah, that's the second option, but I'd be really happy with the checkboxes sorted alphabetically. Most users will only check a few I think, so it's not a hard process. Besides, I'm not sure I want to introduce all those dependencies.

**Comment by icholy** [comment]
Ok, I'll need to change/extend how options are handle.

**Pull request closed without merging**